client/web/web.go

```diff
@@ -456,16 +456,10 @@ var localapiAllowlist = []string{
 // If the server is running in CGI mode, the key is cached to disk and reused between requests.
 // If an error occurs during key storage, the error is logged and the active process terminated.
 func (s *Server) csrfKey() []byte {
-	var csrfFile string
+	csrfFile := filepath.Join(os.TempDir(), "tailscale-web-csrf.key")
```

Comment:
can you easily check if if a /tmp/tailscale-web-csrf.key exists already before creating it?

Reply:
Alternatively I suppose we could use `os.MkdirTemp` to ensure we don't have a collision (though if the idea is to reuse the value over restarts, perhaps not).

Reply:
yeah, exactly... in a CGI environment, every request is a run in a separate process, so we have to have a stable file between runs.

Reply:
outside of CGI, when we actually start up a server that servers requests, we don't cache the csrf key at all... we just hold it in memory.  This is only a workaround for CGI (which proves to be super common with NAS devices... synology, qnap, and unraid all use CGI for plugins)